### PR TITLE
Only show entities associated with selected profile

### PR DESCRIPTION
**Pull request by dippindots**
Fix cBioPortal/cbioportal#8520

If there are more than one profiles falls into one `Generic Assay Type`, in this case, one tab has more than one profiles.
When you switch between profiles, entities should be changed based on your selection. Previously, we just show all entities in all profiles.

image

image

### CODE REVIEW

File: src/pages/studyView/addChartButton/AddChartButton.tsx [hunk 1]
```diff
@@ -99,12 +113,34 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
         super(props, context);
         makeObservable(this);
         this.tabsDivRef = React.createRef<HTMLDivElement>();
+
+        // Using the first profile to initialze genericAssayProfileOptionsByType
+        this.genericAssayEntityOptionsReaction = when(
```
Comment (alisman) [inline src/pages/studyView/addChartButton/AddChartButton.tsx:118]: interesting. haven't seen when used before

File: src/pages/studyView/addChartButton/AddChartButton.tsx [hunk 2]
```diff
@@ -99,12 +113,34 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
         super(props, context);
         makeObservable(this);
         this.tabsDivRef = React.createRef<HTMLDivElement>();
+
+        // Using the first profile to initialze genericAssayProfileOptionsByType
```
Comment (alisman) [inline src/pages/studyView/addChartButton/AddChartButton.tsx:117]: just want to make sure, is it the first profile or entity?

File: src/pages/studyView/addChartButton/AddChartButton.tsx [hunk 3]
```diff
@@ -433,13 +480,31 @@ class AddChartTabs extends React.Component<IAddChartTabsProps, {}> {
...
             (options, type) => {
+                const molecularProfileIdsInType = options.find(
+                    option =>
+                        option.value ===
+                        this.selectedGenericAssayProfileIdByType.get(type)
+                )!.profileIds;
+                const entities = _.reduce(
+                    molecularProfileIdsInType,
+                    (set, profileId) => {
+                        set = new Set([
```
Comment (alisman) [inline src/pages/studyView/addChartButton/AddChartButton.tsx:491]: this seems odd to create a new Set every iteration.  you cannot add them to existing set?

File: src/pages/studyView/addChartButton/AddChartButton.tsx (hunk 2 above)
Comment (dippindots) [reply src/pages/studyView/addChartButton/AddChartButton.tsx:117]: This should be the first profile, we want to generate entity options based on a profile selection, so use the first profile as the default.

**Pull request closed without merging**